src/types/ChannelOptions.ts

```diff
@@ -43,4 +43,12 @@ export interface ChannelOptions {
    * @type {number}
    */
   maxRolesAssigned?: number;
+
+  /**
+   * Private member to store the function for event listener `GuildMemberUpdate`.
+   *
+   * @private
+   * @type {Function}
+   */
+  readonly _rolesChangesListener: (oldMember: GuildMember | PartialGuildMember, newMember: GuildMember) => Awaitable<void>;
```

Comment:
Nor private nor readonly properties can be assigned to an object literal. This kind of access modifiers is only existing in the context of an encapsulation object such as a class, an abstract class or an interface. Thus you won't be able to protect this property from outsiders.

I would not manage the listener with the collection managing the public API. You should probably create a second private collection for internal purposes and store the listener based on the channel id in it.